corehq/apps/reports/static/reports/js/user_history.js

```diff
@@ -0,0 +1,16 @@
+hqDefine("reports/js/user_history", [
+    'jquery',
+], function (
+    $
+) {
+    $(function () {
+        if (document.location.href.match(/reports\/user_history/)) {
```

Comment:
This check is a pattern that's used in other report-specific js, like `inspect_data.js`. I'm not worried about it changing if the URL changes, I think we'd grep for it and if not, it would be apparent that it's broken pretty quickly.

It's true that this class isn't necessarily specific to this report, so it could go in a more generic file. But to avoid bloat, my general principle is to put code in the most specific place possible, and then move it when other code starts using it.

Reply:
> It's true that this class isn't necessarily specific to this report, so it could go in a more generic file. But to avoid bloat, my general principle is to put code in the most specific place possible, and then move it when other code starts using it.

Sorry, i just meant that if the reason for adding a url check was to avoid conflicting with DOM elements with same class/Id on a different page, we could just keep the class/id naming very specific and then not need to add URL check at all.

Reply:
Going to leave this as-is - I know it would work either way, but the URL checking is a pattern I want to reinforce.